modules/promotion/src/PromotionStorage.php

```diff
@@ -107,6 +107,46 @@ public function loadByCoupon(OrderTypeInterface $order_type, StoreInterface $sto
   }
 
   /**
+   * Builds a query that will load all promotions that are no longer valid
+   * determined by the base field limiting values.
+   *
+   * @param bool $only_enabled
+   *   Only include currently enabled promotions that have expired.
+   * @return array|bool|\Drupal\Core\Entity\EntityInterface[]
+   *   The expired promotion entities. Returns FALSE if none found.
+   */
+  public function loadExpired($only_enabled = TRUE) {
+    // Subquery to determine the amount of times a coupon has been used.
+    $usage_query = $this->database->select('commerce_promotion_usage', 'cpu');
+    $usage_query->addExpression('COUNT(cpu.promotion_id)', 'count');
+    $usage_query->where('cpu.promotion_id = cpd.promotion_id');
+    $usage_query->groupBy('cpu.promotion_id');
```

Comment:
Would it work if I add another method to PromotionStorage that will load and check promotion uses by calling getUsageMultiple? That way I can run a quick query to only pass promotions to it which are active and have a usage set.

I think the only way I can mess around with getUsageMultiple to handle return usage on its own would be to be just make it load all promotions if none are passed which could be excessive even if we're not too concerned w/ performance.

Thoughts?